plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/DecoratedHiveMetastoreModule.java

```diff
@@ -56,7 +68,9 @@ protected void setup(Binder binder)
         newExporter(binder).export(HiveMetastoreFactory.class)
                 .as(generator -> generator.generatedNameOf(CachingHiveMetastore.class));
 
-        newSetBinder(binder, Procedure.class).addBinding().toProvider(FlushHiveMetastoreCacheProcedure.class).in(Scopes.SINGLETON);
+        if (installFlushMetadataCacheProcedure) {
```

Comment:
Do we want this procedure installed for `iceberg` ? 
tbh, I didn't know that it existed in iceberg already.

Reply:
I'd rather have the method always available in the connector (register it on `HiveProcedureModule`) and throw an exception (as your implementation does for delta) in case that the caching metastore is not used.

Reply:
Added another commit to avoid installing the procedure in Iceberg.

> I'd rather have the method always available in the connector 

What's the rationale of this? My opinion is basically opposite.

Reply:
> What's the rationale or motivation of this?

I was thinking more from a user perspective where a user may be confused that the procedure is not available (e.g. : while using AWS Glue metastore) for the hive/delta connector even though is advertised to be available in the documentation.